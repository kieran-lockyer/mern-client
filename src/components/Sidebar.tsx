import React from "react";
import { Icon } from "@blueprintjs/core";
import { NavLink } from "react-router-dom";
import styled from "styled-components";

const Sidebar = () => (
  <Navbar>
    <NavbarBrand>
      <img src="/images/logo.png" alt="Brand Logo" />
    </NavbarBrand>

    <NavbarNav>
      <NavItems>
        <NavItem exact to="/">
          <Icon icon="dashboard" iconSize={25} />
        </NavItem>
      </NavItems>
      <NavItems>
        <NavItem to="/photos">
          <Icon icon="media" iconSize={25} />
        </NavItem>
      </NavItems>
      <NavItems>
        <NavItem to="/tags">
          <Icon icon="tag" iconSize={25} />
        </NavItem>
      </NavItems>
      <NavItems>
        <NavItem to="/settings">
          <Icon icon="cog" iconSize={25} />
        </NavItem>
      </NavItems>
    </NavbarNav>
  </Navbar>
);

const Navbar = styled.div`
  display: flex;
  flex: 1 0 100%;
  flex-direction: column;
  align-items: center;
  max-width: 5rem;
  background: #172336;
  margin-right: auto;
  @media only screen and (max-width: 800px) {
    flex-direction: row;
    width: 100%;
    max-width: 100%;
  }
`;

const NavbarBrand = styled.div`
  margin: 3rem auto;

  @media only screen and (max-width: 800px) {
    margin 0;
  }


  & img {
    border-radius: 50%;
    width: 100%;
    height: auto;
    padding: 0.7rem;
    max-height: 70px;
    width: auto;
  }
`;

const NavbarNav = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  flex-basis: 100%;
  @media only screen and (max-width: 800px) {
    flex-direction: row;
  }
`;

const NavItems = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  transition: all 1s ease-in;
`;

const NavItem = styled(NavLink)`
  width: 100%;
  min-height: 4rem;
  display: flex;
  justify-content: center;
  flex: 0 0 2rem;
  align-items: center;
  transition: all 0.2s;
  color: #fff;

  & svg {
    fill: #fff !important;
  }

  &:hover,
  &:active {
    background: #48c0b9;
  }

  &.${props => props.activeClassName} {
    background: #48c0b9;
  }
`;

NavItem.defaultProps = {
  activeClassName: "active"
};

export default Sidebar;
