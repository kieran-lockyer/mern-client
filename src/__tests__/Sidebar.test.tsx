import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from '../components/Sidebar';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<Sidebar />', () => {
  describe('render()', () => {
    test('renders the sidebar component', () => {
      const wrapper = shallow(<Sidebar />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});