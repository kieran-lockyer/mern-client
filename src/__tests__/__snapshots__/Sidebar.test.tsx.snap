// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> render() renders the component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex: 1 0 100%;
  flex-direction: column;
  align-items: center;
  max-width: 5rem;
  background: #172336;
  margin-right: auto;
  @media only screen and (max-width: 800px) {
    flex-direction: row;
    width: 100%;
    max-width: 100%;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <styled.div>
    <img
      alt="Brand Logo"
      src="/images/logo.png"
    />
  </styled.div>
  <styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        exact={true}
        to="/"
      >
        <Blueprint3.Icon
          icon="dashboard"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        to="/photos"
      >
        <Blueprint3.Icon
          icon="media"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        to="/tags"
      >
        <Blueprint3.Icon
          icon="tag"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        to="/settings"
      >
        <Blueprint3.Icon
          icon="cog"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
  </styled.div>
</StyledComponent>
`;

exports[`<Sidebar /> render() renders the sidebar component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex: 1 0 100%;
  flex-direction: column;
  align-items: center;
  max-width: 5rem;
  background: #172336;
  margin-right: auto;
  @media only screen and (max-width: 800px) {
    flex-direction: row;
    width: 100%;
    max-width: 100%;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <styled.div>
    <img
      alt="Brand Logo"
      src="/images/logo.png"
    />
  </styled.div>
  <styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        exact={true}
        to="/"
      >
        <Blueprint3.Icon
          icon="dashboard"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        to="/photos"
      >
        <Blueprint3.Icon
          icon="media"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        to="/tags"
      >
        <Blueprint3.Icon
          icon="tag"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
    <styled.div>
      <Styled(NavLink)
        activeClassName="active"
        to="/settings"
      >
        <Blueprint3.Icon
          icon="cog"
          iconSize={25}
        />
      </Styled(NavLink)>
    </styled.div>
  </styled.div>
</StyledComponent>
`;
